File: Data Structures & Algorithms/binary-search/submission-3.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l = 0
        r = len(nums) - 1
        mid = (l + r) // 2

        while l <= r:
            if target == nums[mid]:
                return mid
            if target == nums[l]:
                return l
            if target == nums[r]:
                return r
            if mid == l or mid == r:
                return -1
            elif target < nums[mid]:
                r = mid
                mid = (l + r) // 2
            else:
                l = mid
                mid = (l + r) // 2
        return -1

File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        
        let count = {}

        for(const num of nums) {
            count[num] = (count[num] || 0) + 1
        }
        
        const maxHeap = new MaxPriorityQueue((n) => n[1])

        for(const [num, cnt] of Object.entries(count)) {
            maxHeap.enqueue([num, cnt])
        }
        let res = []
        for(let i = 0; i < k; ++i) {
            const [num, cnt] = maxHeap.dequeue()
            res.push(num)
        }
        return res
    }
}
